frontend/packages/component-library/src/carousel/Carousel.tsx

```diff
@@ -0,0 +1,138 @@
+import classNames from 'classnames';
+import {HTMLAttributes, ReactNode} from 'react';
+
+// Import Swiper React components
+// See Swiper documentation here: https://swiperjs.com/react
+import {Swiper, SwiperSlide} from 'swiper/react';
+import {Navigation, Pagination, A11y} from 'swiper/modules';
+
+// Import Swiper styles
+import 'swiper/css';
+import 'swiper/css/navigation';
+import 'swiper/css/pagination';
+
+import moduleStyles from './carousel.module.scss';
+import FontAwesomeV6Icon from '@/fontAwesomeV6Icon/FontAwesomeV6Icon';
+
+export interface CarouselProps extends HTMLAttributes<HTMLElement> {
+  /** Unique className (required) */
+  uniqueClassName: string;
```

Comment:
Rather than using a `uniqueClassName`, I wonder if it would be better to use `const uuid = crypto.randomUUID()` within the component itself. This would guarantee that the buttons are only used within the component with uniqueness so in the case there are multiple carousels it wouldn't matter. The main issue I possibly see is if multiple carousels are dragged in and somehow the prop isn't unique causing the buttons to manipulate different carousels that the intended.

Reply:
Love this idea, updated! (Still figuring out the console errors re: the slide keys though.)